litgpt/deploy/serve.py

```diff
@@ -129,6 +132,61 @@ def encode_response(self, output):
             yield {"output": out}
 
 
+class OpenAISpecLitAPI(BaseLitAPI):
+    def __init__(
+        self,
+        checkpoint_dir: Path,
+        quantize: Optional[str] = None,
+        precision: Optional[str] = None,
+        temperature: float = 0.8,
+        top_k: int = 50,
+        top_p: float = 1.0,
+        max_new_tokens: int = 50,
+        devices: int = 1,
+    ):
+        super().__init__(checkpoint_dir, quantize, precision, temperature, top_k, top_p, max_new_tokens, devices)
+
+    def setup(self, device: str):
+        super().setup(device)
+
+        config_path = self.checkpoint_dir / "tokenizer_config.json"
+        if not config_path.is_file():
+            raise FileNotFoundError(f"Tokenizer config file not found at {config_path}")
+
+        with open(config_path, encoding="utf-8") as fp:
+            config = json.load(fp)
+            chat_template = config.get("chat_template", None)
+            if chat_template is None:
+                raise ValueError("chat_template not found in tokenizer config file.")
+            self.chat_template = chat_template
+
+    def decode_request(self, request: "ChatCompletionRequest") -> Any:
+        prompt = self.apply_chat_template(request.messages)
```

Comment:
this will create the `Template` object on every new request. Since we have a single chat template, we should store it as a property.

Reply:
Ah, got it. Thanks for pointing that out, @aniketmaurya! 🙌